eras/alonzo/impl/cddl/lib/Cardano/Ledger/Alonzo/HuddleSpec.hs

```diff
@@ -0,0 +1,589 @@
+
+instance HuddleGroup "script_all" AlonzoEra where
+  huddleGroup = scriptAllGroup @AlonzoEra
+
+instance HuddleGroup "script_any" AlonzoEra where
+  huddleGroup = scriptAnyGroup @AlonzoEra
+
+instance HuddleGroup "script_n_of_k" AlonzoEra where
+  huddleGroup = scriptNOfKGroup @AlonzoEra
+
+instance HuddleGroup "script_invalid_before" AlonzoEra where
+  huddleGroup = scriptInvalidBeforeGroup @AlonzoEra
+
+instance HuddleGroup "script_invalid_hereafter" AlonzoEra where
+  huddleGroup = scriptInvalidHereafterGroup @AlonzoEra
+
+instance HuddleRule "policy_id" AlonzoEra where
+  huddleRule p = "policy_id" =:= huddleRule @"script_hash" p
+
+instance HuddleRule "asset_name" AlonzoEra where
+  huddleRule _ = "asset_name" =:= VBytes `sized` (0 :: Word64, 32 :: Word64)
+
+instance HuddleRule "native_script" AlonzoEra where
+  huddleRule = nativeScriptRule @AlonzoEra
+
+instance HuddleRule "value" AlonzoEra where
+  huddleRule p =
+    "value"
+      =:= huddleRule @"coin" p
+      / sarr [a $ huddleRule @"coin" p, a $ multiasset p VUInt]
+
+instance HuddleRule "mint" AlonzoEra where
+  huddleRule p = "mint" =:= multiasset p (huddleRule @"int64" p)
+
+instance HuddleRule "block" AlonzoEra where
+  huddleRule p =
+    comment
+      [str|Valid blocks must also satisfy the following two constraints:
+          |  1) the length of transaction_bodies and transaction_witness_sets must be
+          |     the same
+          |  2) every transaction_index must be strictly smaller than the length of
+          |     transaction_bodies
+          |NEW:
+          |  invalid_transactions
+          |]
+      $ "block"
+        =:= arr
+          [ a $ huddleRule @"header" p
+          , "transaction_bodies" ==> arr [0 <+ a (huddleRule @"transaction_body" p)]
+          , "transaction_witness_sets" ==> arr [0 <+ a (huddleRule @"transaction_witness_set" p)]
+          , "auxiliary_data_set"
+              ==> mp
+                [ 0
+                    <+ asKey (huddleRule @"transaction_index" p)
+                    ==> huddleRule @"auxiliary_data" p
+                ]
+          , "invalid_transactions" ==> arr [0 <+ a (huddleRule @"transaction_index" p)]
+          ]
+
+instance HuddleRule "header" AlonzoEra where
+  huddleRule p =
+    "header"
+      =:= arr
+        [ a $ huddleRule @"header_body" p
+        , "body_signature" ==> huddleRule @"kes_signature" p
+        ]
+
+instance HuddleRule "header_body" AlonzoEra where
+  huddleRule p =
+    "header_body"
+      =:= arr
+        [ "block_number" ==> huddleRule @"block_number" p
+        , "slot" ==> huddleRule @"slot" p
+        , "prev_hash" ==> (huddleRule @"hash32" p / VNil)
+        , "issuer_vkey" ==> huddleRule @"vkey" p
+        , "vrf_vkey" ==> huddleRule @"vrf_vkey" p
+        , "nonce_vrf" ==> huddleRule @"vrf_cert" p
+        , "leader_vrf" ==> huddleRule @"vrf_cert" p
+        , "block_body_size" ==> VUInt //- "merkle triple root"
+        , "block_body_hash" ==> huddleRule @"hash32" p
```

Comment:
I believe that comment was intended for the block body hash, instead of size
```suggestion
        , "block_body_size" ==> VUInt
        , "block_body_hash" ==> huddleRule @"hash32" p //- "merkle triple root"
```